### ISSUE TITLE
[BUG] ISO installation console can't correctly reflect the Harvester provisioning status

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#6690]
**Describe the bug**

Use ISO installation to provision Harvester, the console page always keep in `Setting up Harvester` and `Setting up node`

  ![image](https://github.com/user-attachments/assets/1f9d1bfe-d8f3-4ee3-aae2-39c4fdc86758)

* Actually we can access Harvester dashboard
  ![image](https://github.com/user-attachments/assets/c0ef02be-1d3c-4d68-9e5f-4154ef9e9b2c)

* The node status is Ready
  ```
  harv14-241001:~ # kubectl get nodes
  NAME            STATUS   ROLES                       AGE     VERSION
  harv14-241001   Ready    control-plane,etcd,master   3h29m   v1.29.9+rke2r1
  
  ```

* Pods are running well too
  ![image](https://github.com/user-attachments/assets/5555182b-12db-4607-b47d-1968ae1c6e4e)

**To Reproduce**
1. Use ISO install to provision Harvester
2. Check the console page 
3. Access Harvester dashboard
4. Access Harvester node

**Expected behavior**

The ISO install console should display the correct Harvester cluster and node status
Should not always keep in `Setting up Harvester` and `Setting up node`

**Support bundle**

[supportbundle_da5ad14e-01a6-4d88-b96f-bd3a56a486b0_2024-10-01T14-58-34Z.zip](https://github.com/user-attachments/files/17211745/supportbundle_da5ad14e-01a6-4d88-b96f-bd3a56a486b0_2024-10-01T14-58-34Z.zip)

**Environment**
 - Harvester ISO version: `v1.4-047b0468`
 - Underlying Infrastructure: Single node kvm machine

**Comment by Yu-Jack** [comment]
That status is from these two conditions https://github.com/harvester/harvester-installer/blob/f6d189bfc828879096e28786cfd175ec5a439bab/pkg/console/dashboard_panels.go#L519-L524, which is from these two commands:

1. `kubectl get no -o jsonpath='{.items[*].metadata.name}'`
2. `kubectl -n fleet-local get ManagedChart harvester -o jsonpath='{.status.conditions}' | jq 'map(select(.type == "Processed" and .status == "True")) | length'`

BTW, it also shows `Setting up xxx`  in my local. It's okay with first command, but it showed `0` with second command.

```
kubectl -n fleet-local get ManagedChart harvester -o jsonpath='{.status.conditions}
=> [{"lastUpdateTime":"2024-10-01T04:21:09Z","status":"True","type":"Ready"},{"lastUpdateTime":"2024-10-02T08:13:47Z","status":"True","type":"Defined"}]

kubectl -n fleet-local get ManagedChart harvester -o jsonpath='{.status.conditions}' | jq 'map(select(.type == "Processed" and .status == "True")) | length'
=> 0
```

That value should be greater or equal to 1 in https://github.com/harvester/harvester-installer/blob/f6d189bfc828879096e28786cfd175ec5a439bab/pkg/console/dashboard_panels.go#L467-L473. 

That means we lack one status called `Processed`. I'm not sure why it's missing. However, what I observed is all `ManagedChats` don't have that status in v1.4, but in v1.3.2. 

@bk201 Do you might know something about this?

**Comment by bk201** [comment]
Maybe it's related to we bump rancher to 2.9, @FrankYang0529 please take a look

**Comment by irishgordo** [comment]
This is also present in v1.4.0-rc1

![Screenshot from 2024-10-02 12-26-57](https://github.com/user-attachments/assets/4e32d9dc-1c40-4717-956a-0200db852a94)

**Comment by Yu-Jack** [comment]
Continue to provide more context.

In [rancher/rancher v2.9.2 (In Harvester v1.4)](https://github.com/rancher/rancher/blob/3da2ae0cabd11aa3af8bc463f5e5e74a3f156c71/pkg/controllers/provisioningv2/managedchart/managedchart.go#L166-L176) and [rancher/rancher v2.8.5 (In Harvester 1.3.2)](https://github.com/rancher/rancher/blob/7af1354e9b8900cec6e5360787e291acd376e6d3/pkg/controllers/provisioningv2/managedchart/managedchart.go#L166-L176), both synced the `bundle` status to `managedchart` staus.

I found that rancher/rancher v2.8.5 uses rancher/fleet v0.9.5, and rancher/rancher v2.9.2 uses rancher/fleet v0.10.2.

The rancher/fleet v.0.9.5 sets the `Processed` status in https://github.com/rancher/fleet/blob/e06bc0170211485edded240c6260f339073d8018/internal/cmd/controller/controllers/bundle/controller.go#L66-L75.

The rancher/fleet v0.10.2 doesn't use this `RegisterBundleGeneratingHandler` function anyone. It only used `SetReadyConditions` in https://github.com/rancher/fleet/blob/decbf9ae72eff096b2bee5efe05a92fdcbd5627d/internal/cmd/controller/reconciler/bundle_controller.go#L250

It seems we should check other status fields (Like `Ready`) instead of `Processed`. https://github.com/rancher/fleet/pull/2018 doesn’t have too much information about this.

**Comment by w13915984028** [comment]
I added an refactor enhancement https://github.com/harvester/harvester/issues/6719, which was also requested before about an API endpoint to get the Harvester cluster information and status.

**Comment by TachunLin** [comment]
@w13915984028, 
Just want to confirm should we need to wait for the enhancement https://github.com/harvester/harvester/issues/6719 to be merged to verify this issue. Many thanks.

**Comment by w13915984028** [comment]
@TachunLin No, the #6719 is not planned yet, PR https://github.com/harvester/harvester-installer/pull/858 has fixed this issue and it can be verified on v1.4.0, thanks.

**Comment by TachunLin** [comment]
Verified fixed on `v1.4-a14e133-head` (24/10/07). Close this issue.

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Create mode installation $~~$</summary>

  1. The ISO install console can display the correct Harvester cluster and node status, and can access Harvester dashboard well

  ![image](https://github.com/user-attachments/assets/43717ee9-85f7-425e-87f5-f769f4a4f011)
  
  ![image](https://github.com/user-attachments/assets/ecf78cf8-ba78-45ee-8837-9c0904849e6f)

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Join mode installation $~~$</summary>

  1. The ISO install console can display the correct Harvester cluster and node status, and can access Harvester dashboard well

  ![image](https://github.com/user-attachments/assets/c1c90c6b-783c-4d8c-a422-827ce0f41aa6)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Binary mode installation $~~$</summary>

  1. The ISO install console can display the correct Harvester cluster and node status, and can access Harvester dashboard well

  ![image](https://github.com/user-attachments/assets/536d7545-b160-4dcd-93bc-08a485d491bc)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Join mode in witness role installation $~~$</summary>

  1. The ISO install console can display the correct Harvester cluster and node status, and can access Harvester dashboard well

  ![image](https://github.com/user-attachments/assets/dc0cb331-4277-45f2-8607-79be47b692d4)


 </details>

### Test Information
* Test Environment: Single and two nodes harvester on kvm machines
* Harvester version: `v1.4-a14e133-head` (24/10/07)

### Verify Steps
<details><summary>Create mode installation</summary>

  1. Trigger ISO installation
  2. Select create mode 
  3. Complete the installation process
  4. Check the console installation status
  5. Check can access the Harvester dashboard

</details>


<details><summary>Join mode installation</summary>

  1. Already have a management role cluster
  2. Trigger ISO installation
  3. Select join mode 
  4. Complete the installation process
  5. Check the console installation status
  6. Check can access the Harvester dashboard

</details>

<details><summary>Binary mode installation</summary>

  1. Already have a management role cluster
  2. Trigger ISO installation
  3. Select binary mode 
  4. Complete the installation process
  5. Check the console installation status
  6. Check can access the Harvester dashboard

</details>

<details><summary>Join mode in witness role installation</summary>

  1. Already have a management role cluster
  2. Trigger ISO installation
  3. Select join mode 
  4. Select the witness role
  5. Complete the installation process
  6. Check the console installation status
  7. Check can access the Harvester dashboard

</details>
